src/library/scala/PartialFunction.scala

```diff
@@ -245,7 +245,10 @@ object PartialFunction {
   /** Converts ordinary function to partial one
    *  @since   2.10
    */
-  def apply[A, B](f: A => B): PartialFunction[A, B] = { case x => f(x) }
+  def apply[A, B](f: A => B): PartialFunction[A, B] = f match {
+    case pf: PartialFunction[A, B] => pf
```

Comment:
I don't think this case will trigger, because the compiler only instantiates an instance of `PartialFunction[A,B]` when the expected type is `PartialFunction[A,B]`. So this will only help for 
```
val pf: PartialFunction[Int,String] = { case 1 => "1" }
val f = PartialFunction(pf)
```